### ISSUE TITLE
Figure out optimal CLI commands

### ISSUE THREAD

**Issue by pdehaan** [rpappalax/ff-tool#5]
OK, the way I see it, we have 4 major commands (with `profile` having 2 main options):
```sh
$ ff download [-c {channel}]
$ ff install [-c {channel}]
$ ff uninstall [-c {channel}]
$ ff profile --create {name}
$ ff profile --delete {name}
```

I'm not 100% sold on how `ff profile -c/--create` should work though. Currently when creating a profile, we need the following bits of information:
```py
def create_mozprofile(application, test_type, env, profile_dir):
```

Where, the args are as follows:
- `application`: name of an application, ie: "loop-server"
- `test_type`: directory of the test type we want to run, ie: "stack-check"
- `env`: testing environment we plan on using, ie: "dev", "stage", or "prod" (or something else)
- `profile_dir`: name of the directory to save the profile to (which will have a prefs.js and user.js -- or something close to that).

We have those first 3 args, so we can do something like load the following config files:
- "prefs.ini"
- "{application}/prefs.ini:{env}"
- "{application}/{test_type}/prefs.ini:{env}"

I don't know how we want to pass those 4 args.

Not sure if we need to add one more nested depth of commands for `ff profile` so it's something like:
```sh
$ ff profile create --application loop-server --test_type stack-check --env stage --name rpapa9000
```

otherwise, we may be able to do something like this:
```sh
$ ff profile --create rpapa9000 --application loop-server --test_type stack-check --env stage
# ...
$ ff profile --delete rpapa9000
```

Not sure if that is awkward since we're treating `--create` and `--delete` as sub-subparsers, but binding the profile name/directory to it.

If we go with the latter option of `ff profile --delete {name}`, not sure how awkward it may be if `ff profile --help` gives you a bunch of non-relevant CLI options for `--delete` when we really only want the name of the profile to delete.

DISCUSS!

**Comment by rpappalax** [comment]
This is freakin great Peter.  Thanks for looking so closely at the options. 
I like the "otherwise" option.  Going more than one level deep on subparsers gonna make u crazy.
I was also thinking about the most common use case is going to be:
I want to "launch" a nightly browser with a profile named "pancakes1".  I should be able to do that easily with a one-liner.  And.. If I don't have a Nightly, ff will go fetch it.  If I don't have a profile named pancakes1, ff will create it -or- if I do, it will make use of it.  Simpler testing, right?
In other words, we should add launch or "go" to the menu.  Normally we do this from the GUI, get the bogus profile-launcher menu that is sometimes crunched really small because you have an external monitor connected.... this way we avoid all that and stay high n dry with the cli.

**Comment by pdehaan** [comment]
I'm not sure how to create a "pancakes1" profile. I mean, I can make you a generic profile with the basic/default options, but it wouldn't be specific for Loop or e2e tests or anything else (since we'd need to know the application and test_type for that).

**Comment by rpappalax** [comment]
Yes, that's what I imagined too.
I think the strength of the mozxxxx libs is that they allow you to specify an incredible number of options for things like releases.  But our most common use-case is much simpler.  Something like: "hey, I just want to run a test on Nightly with a new profile."  If we wanted to do that with mozxxx, we literally had to Google how to do that.  So, what we're providing with this wrapper is a bunch of defaults (which I believe you've already put in place).
Ex.:  If I try to launch a nightly browser using ff with a profile name to "pancakes1" and that profile doesn't exist, then we just use the global prefs (which would be default).
What do you think?

**Comment by pdehaan** [comment]
#### Steps:
1. Download Y/N? (always download &mdash; mozdownload handles caching)
2. Install Y/N? (always reinstall? find out if we try and do max 1x daily)
3. Profile exists Y/N? (if Y, use specified profile. if N generate profile w/ specified name)
4. Launch

---

#### CLI Options:
1. `ff` &mdash; download Nightly (if doesn't exist) and install it and generate random profile and launch Firefox.
2. `ff -c aurora` &mdash; same as above, but w/ Aurora/DeveloperEdition.
3. `ff -c beta -p rpapa1` &mdash; same as above, but w/ Beta and a named profile (which we create if it doesn't exist).
4. `ff -c beta -p rpapa1` &mdash; same as above, but we don't recreate profile because it exists. We may have to redownload and will reinstall Beta channel since it may have updated since last run.


**TODO:** Figure out if mozdownload has ability to determine if newer build was downloaded so we can hook that into our installer script. So we only install if newer bits were downloaded.

**Comment by pdehaan** [comment]
1. download, install, profile, launch
`$ ff`

2. download, install, ~~profile~~, ~~launch~~
`$ ff --refresh-only [-c CHANNEL]`
